django/contrib/postgres/fields/ranges.py

```diff
@@ -199,23 +199,32 @@ def as_sql(self, compiler, connection):
 class RangeContainedBy(lookups.PostgresSimpleLookup):
     lookup_name = 'contained_by'
     type_mapping = {
+        'smallserial': 'int4range',
+        'serial': 'int4range',
+        'bigserial': 'int8range',
+        'smallint': 'int4range',
         'integer': 'int4range',
         'bigint': 'int8range',
         'double precision': 'numrange',
+        'numeric': 'numrange',
         'date': 'daterange',
         'timestamp with time zone': 'tstzrange',
     }
     operator = RangeOperators.CONTAINED_BY
 
     def process_rhs(self, compiler, connection):
         rhs, rhs_params = super().process_rhs(compiler, connection)
-        cast_type = self.type_mapping[self.lhs.output_field.db_type(connection)]
+        #  Split by `(` because for DecimalField we have numeric(x, y)
+        db_type = self.lhs.output_field.db_type(connection).split('(')[0]
+        cast_type = self.type_mapping[db_type]
         return '%s::%s' % (rhs, cast_type), rhs_params
 
     def process_lhs(self, compiler, connection):
         lhs, lhs_params = super().process_lhs(compiler, connection)
         if isinstance(self.lhs.output_field, models.FloatField):
             lhs = '%s::numeric' % lhs
+        elif isinstance(self.lhs.output_field, (models.SmallAutoField, models.SmallIntegerField)):
```

Comment:
Since `SmallAutoField` extends `SmallIntegerField` this can be reduced to

```suggestion
        elif isinstance(self.lhs.output_field, models.SmallIntegerField):
```